src/main/java/com/yelp/nrtsearch/server/luceneserver/IndexState.java

```diff
@@ -133,6 +134,7 @@
 
   private static final Pattern reSimpleName = Pattern.compile("^[a-zA-Z_][a-zA-Z_0-9]*$");
   private ThreadPoolExecutor searchThreadPoolExecutor;
+  private IdFieldDef idFieldDef = null;
```

Comment:
This may need to be declared volatile (or AtomicReference) since getIdFieldDef is unsynchronized, and threads could technically miss the update. I don't think there will be a big performance hit, since there will be no writes in stead state.

I'm not sure if this would be a problem in practice. If this seems like overkill, let me know.

Reply:
Yeah, I don't think that should be a problem as the idFieldDef change is only going to happen during startup or adding new fields. It will be a while before it's actually used via new indexing requests. If there are incoming requests which don't have the idField values, it will break as expected.

Reply:
As long as it doesn't get into the thread cache somehow before being set, that should be ok.

Reply:
Since there is one index state per-process per-index and we create new callables while adding new documents, I think it should be okay for now.